subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputHistoryLossIntegrationTest.groovy

```diff
@@ -142,7 +139,8 @@ class StaleOutputHistoryLossIntegrationTest extends AbstractIntegrationSpec {
             
             task configureCompileJava {
                 doLast {
-                    sourceSets.main.output.classesDir = file('out')
+                    compileJava.destinationDir = file('build/out')
+                    jar.from compileJava
```

Comment:
The test was ignored (`NotYetImplemented`). Now I wanted it to pass. The test was actually failing since calling `classesDir = ...` caused a deprecation message. I wanted to get rid of the deprecation message and therefore I did this change.